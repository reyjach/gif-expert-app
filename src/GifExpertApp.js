import React, { useState } from 'react'
import { AppCategory } from './components/AppCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    //const categories = ['Hola','Minecraft','Google'];
    const [categories, setCategories] = useState(['Minecraft']);


    return (
        <>
            <h2>GifExpertApp</h2>
            <AppCategory setCategories={ setCategories }/>
            <hr />
            <ol>
                { categories.map( category => (
                    <GifGrid 
                        category = {category}
                        key={category}
                    />
                ))}
            </ol>
        </>
    )
}

export default GifExpertApp

