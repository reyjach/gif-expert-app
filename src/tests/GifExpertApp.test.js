import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Probar el componente GifExpertApp', () => { 


    test('Debe de mostrar el componente', () => { 

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de categorias', () => { 

        const categories = ['Goku','auronplay'];

        const wrapper = shallow( <GifExpertApp defauldCategories={categories}/> );

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('GifGrid').length ).toBe(categories.length)
     })
 })