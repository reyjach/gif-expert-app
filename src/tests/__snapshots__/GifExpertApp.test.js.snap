// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente GifExpertApp Debe de mostrar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AppCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probar el componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AppCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="auronplay"
      key="auronplay"
    />
  </ol>
</Fragment>
`;
